LiteCore/Query/N1QL_Parser/n1ql_parser_internal.hh

```diff
@@ -349,6 +358,66 @@ namespace litecore::n1ql {
+    //        "UNION UNIQUE UNKNOWN UNSET UPDATE UPSERT USE USER USERS VALIDATE"
+    //        " VALUE VALUES VECTOR VIA VIEW WHILE WINDOW WITH WITHIN WORK XOR";
+
+    constexpr const char* kServerReservedWords =
+            " ALL          ARRAY        AT        BEGIN      CAST        CORRELATED COVER        CURRENT"
+            " DECREMENT    DEFAULT      DERIVED   DESCRIBE   DO          EACH       ELEMENT      ESCAPE"
+            " EXCEPT       EXCLUDE      EXECUTE   EXISTS     EXPLAIN     FETCH      FILTER       FIRST"
+            " FLATTEN      FLATTEN_KEYS FOLLOWING FOR        FORCE       FUNCTION   GRANT        GROUPS"
+            " HASH         IF           IGNORE    ILIKE      INCLUDE     INCREMENT  INDEX        INFER"
+            " INLINE       INTERSECT    ISOLATION KEY        KEYS        KEYSPACE   KNOWN        LAST"
+            " LATERAL      LET          LETTING   LEVEL      LSM         MAP        MAPPING      MATCHED"
+            " MATERIALIZED MAXVALUE     MERGE     MINUS      MINVALUE    NAMESPACE  NAMESPACE_ID NEST"
+            " EXT          NEXTVAL      NL        NO         NOT_A_TOKEN NTH_VALUE  NULLS        NUMBER"
+            " OBJECT       OPTION       OPTIONS   OTHERS     OVER        PARSE      PARTITION    PASSWORD"
+            " PATH         POOL         PRECEDING PREPARE    PREV        PREVIOUS   PREVVAL      PRIMARY"
+            " PRIVATE      PRIVILEGE    PROBE     PROCEDURE  PUBLIC      RANGE      RAW          READ"
+            " REALM        RECURSIVE    REDUCE    RENAME     REPLACE     RESPECT    RESTART      RESTRICT"
+            " RETURN       RETURNING    REVOKE    ROLE       ROLES       ROLLBACK   ROW          ROWS"
+            " SAVEPOINT    SCHEMA       SCOPE     SELF       SEMI        SEQUENCE   SHOW         SOME"
+            " START        STATISTICS   STRING    SYSTEM     TIES        TO         TRAN         TRIGGER"
+            " TRUNCATE     UNBOUNDED    UNDER     UNION      UNIQUE      UNKNOWN    UNSET        UPDATE"
+            " UPSERT       USE          USER      USERS      VALIDATE    VALUE      VALUES       VECTOR"
+            " VIA          VIEW         WHILE     WINDOW     WITH        WITHIN     WORK         XOR";
+
+    bool isServerReservedWord(std::string word) {
+        static std::unordered_set<std::string_view> serverReserved;
+        if ( serverReserved.empty() ) {
+            const char* p     = kServerReservedWords;
+            const char* start = nullptr;
+            for ( ; *p != '\0'; ++p ) {
+                if ( *p == ' ' ) {
+                    if ( start != nullptr ) {
+                        //start -> p
+                        serverReserved.emplace(start, p);
+                        start = nullptr;
+                    }
+                } else if ( start == nullptr ) {
+                    start = p;
+                }
+            }
+            if ( start != nullptr ) { serverReserved.emplace(start, p); }
```

Comment:
Use `split()` for this instead of nasty C code:
```c++
    /** Splits the string at occurrences of `separator` and calls the callback for each piece.
        There may be empty pieces, if the separator occurs at the start or end or twice in a row. */
    void split(std::string_view str, std::string_view separator, fleece::function_ref<void(std::string_view)> callback);
```
You'll need a consistent delimiter between words though, probably a single space.